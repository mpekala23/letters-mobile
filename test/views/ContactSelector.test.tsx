import * as React from 'react';
import { ContactSelectorScreen } from '@views';
import { render, toJSON, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const setup = (contactsOverrides = []) => {
  const navigation = { navigate: jest.fn(), addListener: jest.fn() };
  const contacts = Object.assign(
    [
      {
        state: 'Minnesota',
        first_name: 'First',
        last_name: 'Last',
        inmate_number: '6',
        relationship: 'Brother',
      },
    ],
    contactsOverrides
  );

  const initialState = {
    adding: {},
    existing: contacts,
  };
  const store = mockStore({
    contact: initialState,
  });

  const StoreProvider = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    navigation,
    store,
    ...render(<ContactSelectorScreen navigation={navigation} contactState={initialState} />, {
      wrapper: StoreProvider,
    }),
  };
};

describe('Contact Selector Screen', () => {
  it('should match snapshot', () => {
    const { container } = setup(<ContactSelectorScreen />);
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should load values for contact cards from the redux store', () => {
    const { getByText } = setup([
      {
        state: 'Minnesota',
        first_name: 'First Contact',
        last_name: 'Contact Last Name',
        inmate_number: '12345678',
        relationship: 'Brother',
      },
      {
        state: 'Minnesota',
        first_name: 'Second Contact',
        last_name: 'Second Contact Last Name',
        inmate_number: '12345678',
        relationship: 'Brother',
      },
    ]);
    expect(getByText('First Contact').props.children).toBe('First Contact');
    expect(getByText('Second Contact').props.children).toBe('Second Contact');
  });

  it('should navigate to contact info screen when the plus button is pressed', () => {
    const { navigation, getByText } = setup();
    fireEvent.press(getByText('+'));
    expect(navigation.navigate).toHaveBeenCalledWith('ContactInfo');
  });
});
