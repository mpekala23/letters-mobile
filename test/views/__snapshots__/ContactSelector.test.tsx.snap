// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Selector Screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#EDEDED",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 16,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "System",
            "fontWeight": "600",
          },
          Object {
            "color": "#6a6a6a",
            "fontSize": 20,
            "paddingBottom": 16,
          },
        ]
      }
    >
      YOUR LOVED ONES
    </Text>
    <ScrollView
      keyboardShouldPersistTaps="handled"
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 135,
            "marginBottom": 16,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": 135,
              "marginBottom": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 16,
                  "paddingTop": 16,
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f6993f",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "width": 80,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22,
                    "fontWeight": "500",
                  }
                }
              >
                FL
              </Text>
            </TouchableOpacity>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingLeft": 112,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#282828",
                    "fontSize": 22,
                    "paddingBottom": 8,
                  },
                ]
              }
            >
              First
            </Text>
            <Text
              style={
                Object {
                  "color": "#6a6a6a",
                  "fontSize": 16,
                  "paddingBottom": 6,
                }
              }
            >
              <span
                aria-label="mail heart emoji"
                role="img"
              >
                üíå
              </span>
               
              received:
            </Text>
            <Text
              style={
                Object {
                  "color": "#6a6a6a",
                  "fontSize": 16,
                  "paddingBottom": 6,
                }
              }
            >
              <span
                aria-label="calender emoji"
                role="img"
              >
                üìÖ
              </span>
               
              last heard from you:
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#6a6a6a",
                    "fontSize": 16,
                    "paddingBottom": 6,
                  },
                  Object {
                    "paddingBottom": 16,
                  },
                ]
              }
            >
              <span
                aria-label="plane emoji"
                role="img"
              >
                ‚úàÔ∏è
              </span>
               
              letters traveled:
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.7}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3490dc",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {},
            Object {
              "alignItems": "center",
              "backgroundColor": "#FC7272",
              "borderRadius": 50,
              "bottom": 0,
              "height": 65,
              "marginBottom": 24,
              "marginRight": 12,
              "paddingLeft": 2,
              "position": "absolute",
              "right": 0,
              "width": 65,
            },
            Object {},
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {
                "fontSize": 30,
              },
              Object {},
            ]
          }
        >
          +
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
